### PR TITLE
Feature/trn 605/e2e core setup

### PR DESCRIPTION
**Pull request by oatymart**
Task: https://oat-sa.atlassian.net/browse/TRN-605

Setup for a new core system for Cypress testing. Contains Cypress dependency, 1 plugin, plugin initialisation, environment config loading mechanism, shareable commands, docs. **No configs, no fixtures, no data, no tests**.

`docs/` - docs
`plugins/` - plugin init + config extender
`support/` - files of common support commands

Test driving this library prototype can be done via installation of https://github.com/oat-sa/tao-deliver-testrunner-nui-fe/pull/161, or by setting it up in your own project and creating a test.

The library should be evaluated against the principles researched in the subpages of https://oat-sa.atlassian.net/wiki/spaces/OAT/pages/852885520/End+to+End+testing+-+Testing+principles

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "bugs": {
+    "url": "https://github.com/oat-sa/e2e-runner/issues"
+  },
+  "homepage": "https://github.com/oat-sa/e2e-runner#readme",
+  "devDependencies": {
+    "@oat-sa/eslint-config-tao": "^1.0.0",
+    "@oat-sa/prettier-config": "^0.1.1",
+    "eslint": "^7.20.0",
+    "eslint-plugin-es": "^4.1.0",
+    "eslint-plugin-jsdoc": "^32.2.0",
```
Comment (krampstudio) [inline package.json:34]: with the last version of `@oat-sa/eslint-config-tao` you don't need to handle plugins yourself

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    extends: "@oat-sa/eslint-config-tao",
```
Comment (krampstudio) [inline .eslintrc.js:2]: suggestion:
we can use the `node` profile and maybe in the future we can add a `cypress` profile in the eslint config
 ```suggestion
    extends: "@oat-sa/eslint-config-tao/node",
```

Comment (krampstudio) [reply .eslintrc.js:2]: and `Cypress` should be in the globals no ?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "@oat-sa-private/e2e-runner",
+  "version": "0.1.0",
+  "displayName": "OAT end-to-end test runner",
+  "description": "End-to-end test runner for OAT applications, based on Cypress",
+  "publishConfig": {
+    "access": "restricted"
```
Comment (krampstudio) [inline package.json:7]: could you please make it public ?

**Review by krampstudio — CHANGES_REQUESTED** [review]
That's good as it is, I've only some remarks and suggestions: 
 - if the shared command file registers itself the commands, would you consider a good practice  to scope or prefix the commands? like `core-login`, etc. to prevent collisions ? 
 -

File: .eslintrc.js (hunk 2 above)
Comment (oatymart) [reply .eslintrc.js:2]: Not 100% sure about using the Node profile. As I understand it, Cypress code is running partly in a Node process, partly in a browser environment. (I set `eslint-config-tao/node`, and had a lint error that `btoa is not defined` (lti.js L73) - as if in Node.js. However it _is_ defined in the test, and produces a valid hash.)

I'm on board for making the custom profile.

**Comment by oatymart** [comment]
> if the shared command file registers itself the commands, would you consider a good practice to scope or prefix the commands? like core-login, etc. to prevent collisions ?

I'm not against that. It makes sense, since the core doesn't know anything about what the consumers will each register.

Didn't have time to add it yet,

File: docs/development.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+Link the package
+
+```sh
+# in e2e-runner
+npm link
+
+# in your project root
+npm link @oat-sa-private/e2e-runner
```
Comment (ampaveliev) [inline docs/development.md:25]:
```suggestion
npm link @oat-sa/e2e-runner
```

**Review by shaveko — APPROVED** [review]
Looks good. Tested this with example tests in https://github.com/oat-sa/tao-deliver-testrunner-nui-fe/pull/161. Works fine

File: .eslintrc.js (hunk 2 above)
Comment (krampstudio) [reply .eslintrc.js:2]: yes, let's check that separately

**Review by krampstudio — APPROVED** [review]
it works well. Eslint can be configured later in the shared repo.

- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] <s>New code is not subject to concurrency issues (if applicable)</s>
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Pull request title and description are meaningful

**Pull request merged**